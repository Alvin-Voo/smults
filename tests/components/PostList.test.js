import React from 'react';
import configureStore from 'redux-mock-store';
import { shallow, mount } from 'enzyme';
import ConnectedPostList, { PostList } from '../../components/PostList';
import postListData from '../fixtures/postList';

const mockStore = configureStore();

describe('Shallow rendering on unconnected PostList', () => {
  test('should render correctly', () => {
    const wrapper = shallow(<PostList posts={postListData} />);
    expect(wrapper).toMatchSnapshot();
  });
});

describe('Test on connected PostList', () => {
  let store;
  let container;

  beforeEach(() => {
    store = mockStore({
      posts: {
        posts: postListData,
      },
    });
    container = mount(<ConnectedPostList store={store} />);
  });

  test('should render correctly', () => {
    expect(container.find('PostList')).toHaveLength(1);
    expect(container.find('Post')).toHaveLength(2);
  });

  test('should have posts prop that matches dummy posts list', () => {
    const postsProp = container.find('PostList').prop('posts');
    expect(postsProp).toHaveLength(2);
    expect(postsProp).toEqual(postListData);
  });
});
