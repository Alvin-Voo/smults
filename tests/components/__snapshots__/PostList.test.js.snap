// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Shallow rendering on unconnected PostList should render correctly 1`] = `
<ItemGroup
  link={true}
>
  <Post
    key="123"
    post={
      Object {
        "author": "dummy",
        "body": "text body",
        "category": "moon",
        "children": "24",
        "created": "5 hours ago",
        "id": 123,
        "image": "http://image.com",
        "net_votes": "213",
        "pending_payout_value": "65.012 SBD",
        "resteemd": undefined,
        "tags": Array [
          "tag1",
          "tag2",
          "tag3",
        ],
        "title": "this is a dummy title",
        "total_payout_value": "0.000 SBD",
        "url": "http://steemit.com/some-permlink",
      }
    }
  />
  <Post
    key="124"
    post={
      Object {
        "author": "dummy2",
        "body": "text body2",
        "category": "sun",
        "children": "12",
        "created": "6 hours ago",
        "id": 124,
        "image": "http://image2.com",
        "net_votes": "2",
        "pending_payout_value": "0.000 SBD",
        "resteemed": true,
        "tags": Array [
          "tag4",
          "tag5",
          "tag6",
        ],
        "title": "this is a title too!",
        "total_payout_value": "105.001 SBD",
        "url": "http://steemit.com/some-permlink-too",
      }
    }
  />
</ItemGroup>
`;
